builder/build_env.py

```diff
@@ -270,7 +321,12 @@ def _create_dep_tree(recipes):
     outputs = []
     for index, recipe in enumerate(recipes):
         deps = []
-        for dep in recipe.get('dependencies', []):
+        dependencies = set()
+        dependencies.update({utils.remove_version(dep) for dep in recipe.get("run_dependencies")})
+        dependencies.update({utils.remove_version(dep) for dep in recipe.get("build_dependencies")})
+        dependencies.update({utils.remove_version(dep) for dep in recipe.get("host_dependencies")})
+        dependencies.update({utils.remove_version(dep) for dep in recipe.get("test_dependencies")})
+        for dep in dependencies:
```

Comment:
@bnemanich 
And then I strip the versions and combine them all into a single set for the dependency tree.